addons/cloudfiles/static/cloudfilesUserConfig.js

```diff
@@ -0,0 +1,173 @@
+/**
+* Module that controls the CloudFiles user settings. Includes Knockout view-model
+* for syncing data.
+*/
+
+var ko = require('knockout');
+var $ = require('jquery');
+var Raven = require('raven-js');
+var bootbox = require('bootbox');
+require('js/osfToggleHeight');
+
+var language = require('js/osfLanguage').Addons.cloudfiles;
+var osfHelpers = require('js/osfHelpers');
+var addonSettings = require('js/addonSettings');
+var ChangeMessageMixin = require('js/changeMessage');
+
+
+var ExternalAccount = addonSettings.ExternalAccount;
+
+var $modal = $('#cloudfilesInputCredentials');
+
+
+function ViewModel(url) {
+    var self = this;
+
+    self.properName = 'Cloud Files';
+    self.username = ko.observable();
+    self.secretKey = ko.observable();
+    self.account_url = '/api/v1/settings/cloudfiles/accounts/';
```

Comment:
This is consistent with how s3 is set up, addonSettings.js isn't a straight forward config file so it's not exactly clear where this would go (at least to me).